### PR TITLE
feat: cm360 transformerproxy V1 features.json update

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> add the new v1 handler for transformer proxy

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request 

## Summary by CodeRabbit

- **New Features**
  - Introduced versioning for network handlers to support multiple API versions.
  - Added new version 'v1' to the list of supported versions for network operations.

- **Enhancements**
  - Updated delivery methods to accept a 'version' parameter for improved service handling.
  - Enhanced error handling with more descriptive messages in network operations.

- **Bug Fixes**
  - Fixed an issue with the delivery controller to correctly handle errors using TypeScript's error variable syntax.

- **Tests**
  - Added tests to ensure network handlers return the correct versions.
  - Implemented new test data for integration with the "campaign_manager" module, covering both successful and failed data deliveries.

### CODE REVIEW

File: src/adapters/networkHandlerFactory.js [hunk 1]
```diff
@@ -16,16 +18,26 @@ SUPPORTED_VERSIONS.forEach((version) => {
...
     } catch {
       // Do nothing as exception indicates
       // network handler is not defined for that destination
     }
   });
 });
 
-const getNetworkHandler = (type) => {
-  const NetworkHandler = handlers[type] || handlers.generic;
+const getNetworkHandler = (type, version) => {
```
Comment (utsabc) [inline src/adapters/networkHandlerFactory.js:39]: Can we add unit test for this  `src/adapters/networkHandlerFactory.js`

File: src/v1/destinations/campaign_manager/networkHandler.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
+const { NetworkError } = require('@rudderstack/integrations-lib');
```
Comment (utsabc) [inline src/v1/destinations/campaign_manager/networkHandler.js:1]: This file is not having any unit/component tests for coverage please add both

File: src/adapters/networkHandlerFactory.js (hunk 1 above)
Comment (aashishmalik) [reply src/adapters/networkHandlerFactory.js:39]: done

File: src/adapters/networkHandlerFactory.js [hunk 3]
```diff
@@ -16,16 +18,26 @@ SUPPORTED_VERSIONS.forEach((version) => {
...
+  const NetworkHandler = handlers[version][type] || handlers.generic;
   return new NetworkHandler();
 };
```
[the remark this answers is not in the document]
Comment (aashishmalik) [reply src/adapters/networkHandlerFactory.js:42]: not required as per internal contract

File: test/integrations/destinations/campaign_manager/dataDelivery/data.ts [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+            FORM: {},
+          },
+          files: {},
+        },
+        method: 'POST',
+      },
+    },
+    output: {
+      response: {
+        status: 500,
```
[the remark this answers is not in the document]
Comment (aashishmalik) [reply test/integrations/destinations/campaign_manager/dataDelivery/data.ts:138]: this case is for entire batch failed .. and is for v0 case ...in V0 we use to retry this code. 
In other PR we have corrected this once I merge both will update test cases too

File: test/integrations/destinations/campaign_manager/dataDelivery/data.ts [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+        },
+      },
+    },
+  },
+  {
+    name: 'campaign_manager',
+    description: 'Failure insert request',
+    feature: 'dataDelivery',
+    module: 'destination',
+    version: 'v0',
```
[the remark this answers is not in the document]
Comment (aashishmalik) [reply test/integrations/destinations/campaign_manager/dataDelivery/data.ts:91]: added

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**